File: ZendAMF/app/servicos/ServicosProduto.php
<?php
//include 'app/db/BaseDados.php';
include 'app/vo/ProdutoVO.php';

class ServicosProduto {
	
	
	private $conn;
	
	function ServicosProduto() {
		$db = new BaseDados();
		$this->conn = $db->conn;
	}
	
	public function addProduto(ProdutoVO $item){
		
		$sql = "insert into produto (codBarra,codGrupo
		 ,descricao ,referencia ,codLocal ,codUnidade
		 ,qtdPorUnidade ,qtdEmEstoque ,codFornecedor ,precoCompra ,precoVenda,foto) values ('{$item->codBarra}' ,{$item->codGrupo} ,'{$item->descricao}'
		 ,'{$item->referencia}' ,{$item->codLocal}
		 ,{$item->codUnidade} ,{$item->qtdPorUnidade}
		 ,{$item->qtdEmEstoque} ,{$item->codFornecedor}
		 ,{$item->precoCompra} ,{$item->precoVenda}
		 ,'{$item->foto}')";
		
		$resultado = $this->conn->Execute($sql);
		$item->codigo = $this->conn->insert_Id();
		return $item;
		
	}
	
	public function removerProduto(ProdutoVO $item){
		
		$sql = "delete from produto  where codigo = {$item->codigo}";
		$resultado = $this->conn->Execute($sql);
		if($this->conn->Affected_Rows()==0){
			return false;
		}
		return true;
	}
	
	public function pesquisarProduto($texto,$coluna){
		$sql = "select * from produto where $coluna like '%$texto%'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new ProdutoVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codBarra = $registro->CODBARRA;
			$dados_item->codGrupo = $registro->CODGRUPO;
			$dados_item->descricao = $registro->DESCRICAO;
			$dados_item->referencia = $registro->REFERENCIA;
			$dados_item->codLocal = $registro->CODLOCAL;
			$dados_item->codUnidade = $registro->CODUNIDADE;
			$dados_item->qtdPorUnidade = $registro->QTDPORUNIDADE;
			$dados_item->qtdEmEstoque = $registro->QTDEMESTOQUE;
			$dados_item->codFornecedor = $registro->CODFORNECEDOR;
			$dados_item->precoCompra = $registro->PRECOCOMPRA;
			$dados_item->precoVenda = $registro->PRECOVENDA;
			$dados_item->foto = $registro->FOTO;
			$retorna_dados_item [] = $dados_item;
		}
		return $retorna_dados_item;
	}
	
	public function getProduto($texto,$coluna){
		$sql = "select * from produto where $coluna = '$texto'";
		$resultado = $this->conn->Execute($sql);
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new ProdutoVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codBarra = $registro->CODBARRA;
			$dados_item->codGrupo = $registro->CODGRUPO;
			$dados_item->descricao = $registro->DESCRICAO;
			$dados_item->referencia = $registro->REFERENCIA;
			$dados_item->codLocal = $registro->CODLOCAL;
			$dados_item->codUnidade = $registro->CODUNIDADE;
			$dados_item->qtdPorUnidade = $registro->QTDPORUNIDADE;
			$dados_item->qtdEmEstoque = $registro->QTDEMESTOQUE;
			$dados_item->codFornecedor = $registro->CODFORNECEDOR;
			$dados_item->precoCompra = $registro->PRECOCOMPRA;
			$dados_item->precoVenda = $registro->PRECOVENDA;
			$dados_item->foto = $registro->FOTO;
		}
		return $dados_item;
	}
	
	public function getProdutos(){
		$sql = "select * from produto";
		
		$resultado = $this->conn->Execute($sql);
		$dados_item = null;
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new ProdutoVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codBarra = $registro->CODBARRA;
			$dados_item->codGrupo = $registro->CODGRUPO;
			$dados_item->descricao = $registro->DESCRICAO;
			$dados_item->referencia = $registro->REFERENCIA;
			$dados_item->codLocal = $registro->CODLOCAL;
			$dados_item->codUnidade = $registro->CODUNIDADE;
			$dados_item->qtdPorUnidade = $registro->QTDPORUNIDADE;
			$dados_item->qtdEmEstoque = $registro->QTDEMESTOQUE;
			$dados_item->codFornecedor = $registro->CODFORNECEDOR;
			$dados_item->precoCompra = $registro->PRECOCOMPRA;
			$dados_item->precoVenda = $registro->PRECOVENDA;
			$dados_item->foto = $registro->FOTO;
			$retorna_dados_item [] = $dados_item;
		}
		return $retorna_dados_item;	
	}
	
	public function filtroProduto($sqlArray){
		$sql = "select * from produto where ";
		$campos = array("descricao","referencia","precoVenda","codFornecedor","codLocal","codUnidade","codGrupo");
		$entrou = false;
		
		
		for ($index = 0; $index < count($sqlArray); $index++) {
			
			if($sqlArray[$index]!= ""){
				if($entrou){
					$sql = $sql." and ".$sqlArray[$index]." ";
				}else{
					$sql = $sql.$sqlArray[$index]." ";
					$entrou = true;
				}
			}
		}
		
		$a = fopen("log.txt","w+");
		fwrite($a,$sql);
		$resultado = $this->conn->Execute($sql);
		$dados_item = null;
		while($registro = $resultado->FetchNextObject()){			
			$dados_item = new ProdutoVO();
			$dados_item->codigo = $registro->CODIGO;
			$dados_item->codBarra = $registro->CODBARRA;
			$dados_item->codGrupo = $registro->CODGRUPO;
			$dados_item->descricao = $registro->DESCRICAO;
			$dados_item->referencia = $registro->REFERENCIA;
			$dados_item->codLocal = $registro->CODLOCAL;
			$dados_item->codUnidade = $registro->CODUNIDADE;
			$dados_item->qtdPorUnidade = $registro->QTDPORUNIDADE;
			$dados_item->qtdEmEstoque = $registro->QTDEMESTOQUE;
			$dados_item->codFornecedor = $registro->CODFORNECEDOR;
			$dados_item->precoCompra = $registro->PRECOCOMPRA;
			$dados_item->precoVenda = $registro->PRECOVENDA;
			$dados_item->foto = $registro->FOTO;
			$retorna_dados_item [] = $dados_item;
		}
		return $retorna_dados_item;	
	}


	public function atualizarProduto(ProdutoVO $produto) {
		
		$sql = "UPDATE produto SET codBarra = '$produto->codBarra', codGrupo = $produto->codGrupo, descricao = '$produto->descricao', referencia = '$produto->referencia',codLocal = $produto->codLocal,codUnidade=$produto->codUnidade,qtdPorUnidade=$produto->qtdPorUnidade,qtdEmEstoque=$produto->qtdEmEstoque,codFornecedor=$produto->codFornecedor,precoCompra=$produto->precoCompra,
		precoVenda=$produto->precoVenda,foto='$produto->foto' where codigo=$produto->codigo";
			
		$resultado = $this->conn->Execute($sql);
		return $produto;	
		
	}
}
//$eu = new ServicosProduto();
//
//$a = array("","","","codFornecedor = '1' ","","codUnidade = '3'");
//
//$eu->filtroProduto($a);

//$c->codBarra = "sdf";
//$c->codFornecedor = 1;
//$c->codGrupo = 1;
//$c->codLocal =1;
//$c->codUnidade = 1;
//$c->descricao = "asdfasdfasdfasdfasdfdafasdfasdfsadf";
//$c->referencia = "asdfkjsaflkasjdfçslkdfjsalçfkj";
//$c->foto = "sdf";
//$c->precoCompra = 1.0;
//$c->precoVenda = 1.0;
//$c->qtdEmEstoque=1;
//$c->qtdPorUnidade = 1;
//$eu->addProduto($c);
//echo $c->codigo;
//$c->codigo = 47;
//$c->dataCadastro = "2010-09-31";
//$eu->getProdutos();

?>

File: trunk/ServidorJava/WEB-INF/src/services/ServicosProduto.java
package services;


import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;

import javax.sql.rowset.spi.SyncResolver;

import vo.ProdutoVO;


public class ServicosProduto {
	
	
	public synchronized ProdutoVO addProduto(ProdutoVO item){
		String sql = "insert into produto (codBarra,codGrupo,descricao ,referencia ,codLocal ,codUnidade ,qtdPorUnidade ,qtdEmEstoque ,codFornecedor ,precoCompra ,precoVenda,foto) values ('"+item.codBarra+"' ,"+item.codGrupo+" ,'"+item.descricao+"'"+
	 ",'"+item.referencia+"' ,"+item.codLocal+""+
		" ,"+item.codUnidade+" ,"+item.qtdPorUnidade+""+
		 ","+item.qtdEmEstoque+" ,"+item.codFornecedor+""+
		 ","+item.precoCompra+" ,"+item.precoVenda+""+
		 ",'"+item.foto+"')";
		Statement st;
		this.banco.conectar();
		this.banco.getConexao().setAutoCommit(false);
		st = this.banco.getConexao().createStatement();
		if (st.executeUpdate(sql, Statement.RETURN_GENERATED_KEYS) == 0) {
			throw new RuntimeException("Erro ao salvar Produto!");
		}
		ResultSet rs = st.getGeneratedKeys();
		if (rs.next())
			item.codigo = rs.getInt(1);
		else
			throw new RuntimeException("Erro ao salvar Produto!");
		this.banco.getConexao().commit();
		return item;
		
	}
	
	public synchronized boolean  removerProduto(ProdutoVO item){
		String sql = "delete from produto  where codigo = "+item.codigo+"";
		if(this.banco.executarNoQuery(sql)==0){
			return false;
		}
		return true;
	}
	
	public synchronized ProdutoVO atualizarProduto(ProdutoVO produto) {
		String sql = "UPDATE produto SET codBarra = '"+produto.codBarra+"', codGrupo = "+produto.codGrupo+", descricao = '"+produto.descricao+"', referencia = '"+produto.referencia+"',codLocal = "+produto.codLocal+",codUnidade="+produto.codUnidade+",qtdPorUnidade="+produto.qtdPorUnidade+",qtdEmEstoque="+produto.qtdEmEstoque+",codFornecedor="+produto.codFornecedor+",precoCompra="+produto.precoCompra+","+
		"precoVenda="+produto.precoVenda+",foto='"+produto.foto+"' where codigo="+produto.codigo+"";
		if(this.banco.executarNoQuery(sql)==0){
			throw new RuntimeException("Erro ao atualizar Produto");
		}
		return produto;	
		
	}
	
	public ArrayList<ProdutoVO> pesquisarProduto(String texto,String coluna){
		String sql = "select * from produto where "+coluna+" like '%"+texto+"%'";
		ResultSet rs =  banco.executar(sql);
		return this.toProdutos(rs);
	}
	
	public ArrayList<ProdutoVO> getProduto(String texto,String coluna){
		String sql = "select * from produto where "+coluna+" = '"+texto+"'";
		ResultSet rs =  banco.executar(sql);
		return  this.toProdutos(rs);
	}
	
	public ArrayList<ProdutoVO> getProdutos(){
		String sql = "select * from produto";
		ResultSet rs =  banco.executar(sql);
		return this.toProdutos(rs);
	}

	public ArrayList<ProdutoVO> getProdutos(String sql){
		ResultSet rs =  banco.executar(sql);
		return this.toProdutos(rs);
	}
	
	public boolean isExiste(ProdutoVO produto) {
		String sql = "select * from produto where codigo = '"
				+ produto.codigo + "'";
		ResultSet rs;
		rs = this.banco.executar(sql);
		if (rs.next())
			return true;
		else
			return false;

	}
	
	private ArrayList<ProdutoVO> toProdutos(ResultSet rs) throws SQLException{
		
		ArrayList<ProdutoVO> produtos = new ArrayList<ProdutoVO>();
		while(rs.next()){
			ProdutoVO dados_item = new ProdutoVO();
			dados_item.codigo = rs.getInt("codigo");
			dados_item.codBarra = rs.getString("codBarra");
			dados_item.codGrupo = rs.getInt("codGrupo");
			dados_item.descricao = rs.getString("descricao");
			dados_item.referencia = rs.getString("referencia");
			dados_item.codLocal = rs.getInt("codLocal");
			dados_item.codUnidade = rs.getInt("codUnidade");
			dados_item.qtdPorUnidade = rs.getInt("qtdPorUnidade");
			dados_item.qtdEmEstoque = rs.getInt("qtdEmEstoque");
			dados_item.codFornecedor = rs.getInt("codFornecedor");
			dados_item.precoCompra = rs.getDouble("precoCompra");
			dados_item.precoVenda = rs.getDouble("precoVenda");
			dados_item.foto = rs.getString("foto");
			produtos.add(dados_item);
		}
		return produtos;
	}	


	
}

